File: palin.py
'''
3
aswini
madam
paap
NO
YES ODD
YES EVEN
'''
n=int(input())
s=['']*n;
c=['']*n;
for i in range(0,n):
    s[i] = str(input())
    temp = s[i]
    c[i] = temp[::-1]
for i in range(0,n):
    if s[i] == c[i]:
        if len(c[i])%2==0:
            print("YES EVEN")
        else:
            print("YES ODD")
    else:
        print("NO")

File: palin.c
#include<stdio.h>
#include<string.h>
 
int main() {
   char str[100],str1[100], temp;
   int t,k;
   int i, j = 0;
   scanf("%d\n",&t);
   while(t-->0)
   {
   gets(str);
   strcpy(str1,str);
   i = 0;
   j = strlen(str) - 1;
 
   while (i < j) {
      temp = str[i];
      str[i] = str[j];
      str[j] = temp;
      i++;
      j--;
   }
 
   if(strcmp(str1,str)==0)
   {
       if(strlen(str)%2==0)
        printf("YES EVEN\n");
       else
        printf("YES ODD\n");
   }
  else
    printf("NO\n");
  }
   return (0);
}


